examples/cpu/par_example.cpp

```diff
@@ -0,0 +1,158 @@
+#include "edm/spacepoint.hpp"
+#include "geometry/pixel_segmentation.hpp"
+#include "algorithms/component_connection.hpp"
+#include "algorithms/measurement_creation.hpp"
+#include "algorithms/spacepoint_formation.hpp"
+#include "csv/csv_io.hpp"
+
+#include <vecmem/memory/host_memory_resource.hpp>
+
+#include <iostream>
+#include <chrono>
+#include "omp.h"
+
+int par_run(const std::string &detector_file, const std::string &cells_dir, unsigned int events) {
+    auto env_d_d = std::getenv("TRACCC_TEST_DATA_DIR");
+    if (env_d_d == nullptr) {
+        throw std::ios_base::failure("Test data directory not found. Please set TRACCC_TEST_DATA_DIR.");
+    }
+    auto data_directory = std::string(env_d_d) + std::string("/");
+
+    // Read the surface transforms
+    std::string io_detector_file = data_directory + detector_file;
+    traccc::surface_reader sreader(io_detector_file,
+                                   {"geometry_id", "cx", "cy", "cz", "rot_xu", "rot_xv", "rot_xw", "rot_zu", "rot_zv",
+                                    "rot_zw"});
+    auto surface_transforms = traccc::read_surfaces(sreader);
+
+    // Algorithms
+    traccc::component_connection cc;
+    traccc::measurement_creation mt;
+    traccc::spacepoint_formation sp;
+
+    // Output stats
+    uint64_t n_cells = 0;
+    uint64_t m_modules = 0;
+    uint64_t n_clusters = 0;
+    uint64_t n_measurements = 0;
+    uint64_t n_space_points = 0;
+
+    // Memory resource used by the EDM.
+    vecmem::host_memory_resource resource;
+
+#pragma omp parallel for reduction (+:n_cells, n_clusters, n_measurements, n_space_points, m_modules)
+    // Loop over events
+    for (unsigned int event = 0; event < events; ++event) {
+
+        // Read the cells from the relevant event file
+        std::string event_string = "000000000";
+        std::string event_number = std::to_string(event);
+        event_string.replace(event_string.size() - event_number.size(), event_number.size(), event_number);
+
+        std::string io_cells_file =
+                data_directory + cells_dir + std::string("/event") + event_string + std::string("-cells.csv");
+        traccc::cell_reader creader(io_cells_file,
+                                    {"geometry_id", "hit_id", "cannel0", "channel1", "activation", "time"});
+        traccc::host_cell_container cells_per_event = traccc::read_cells(creader, resource, surface_transforms);
+        m_modules += cells_per_event.headers.size();
+
+        // Output containers
+        traccc::host_measurement_container measurements_per_event;
+        traccc::host_spacepoint_container spacepoints_per_event;
+        measurements_per_event.headers.reserve(cells_per_event.headers.size());
+        measurements_per_event.items.reserve(cells_per_event.headers.size());
+        spacepoints_per_event.headers.reserve(cells_per_event.headers.size());
+        spacepoints_per_event.items.reserve(cells_per_event.headers.size());
+
+#pragma omp parallel for
```

Comment:
Is your intention here to spawn a nested thread team? I believe you would probably be better off leaving this pragma out and relying only on the outer parallel pragma. I think that would give you better performance.

Reply:
Yes, it was intentionally. On my machine, I saw no performance difference between these 2 choices. But I can double check with much larger number of events and threads and see how that goes.

Reply:
Tested with 1000 parallel events, with and without the nested team, and there was no impact on performance.